pandora/models/pandora.py

```diff
@@ -292,7 +294,18 @@ class SearchResultItem(PandoraModel):
 
     @property
     def is_song(self):
-        return self.song_name is not None
+        # Song result tokens start with 'S', followed by seven digits.
+        return re.compile('^([S])(\d{7})$').match(self.token)
```

Comment:
Would a `startswith` check work just as well here or is the stronger validation needed? If you do want to extra validation then you should just use `re.match('^([S])(\d{7})$', self.token)` since python will memoize the complied regex automatically.